File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/util/TypefaceHelper.java
package com.sorcererxw.matthiasheiderichphotography.util;

import android.content.Context;
import android.graphics.Typeface;
import android.support.annotation.NonNull;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/8/24
 */
public class TypefaceHelper {
    private static Typeface fontDemi;

    private static Typeface fontBook;

    public enum Type {
        Demi,
        Book
    }

    public static
    @NonNull
    Typeface getTypeface(Context context, Type type) {
        switch (type) {
            case Book:
                if (fontBook == null) {
                    fontBook = Typeface.createFromAsset(context.getAssets(), "FuturaPTBook.otf");
                }
                return fontBook;
            case Demi:
                if (fontDemi == null) {
                    fontDemi = Typeface.createFromAsset(context.getAssets(), "FuturaPTDemi.otf");
                }
                return fontDemi;
        }
        return null;
    }
}

File: app/src/main/java/com/sorcererxw/matthiasheiderichphotography/util/TypefaceHelper.kt
package com.sorcererxw.matthiasheiderichphotography.util

import android.content.Context
import android.graphics.Typeface

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/8/24
 */
object TypefaceHelper {
    private var fontDemi: Typeface? = null

    private var fontBook: Typeface? = null

    enum class Type {
        Demi,
        Book
    }

    fun getTypeface(context: Context, type: Type): Typeface? {
        return when (type) {
            TypefaceHelper.Type.Book -> {
                if (fontBook == null) {
                    fontBook = Typeface.createFromAsset(context.assets, "FuturaPTBook.otf")
                }
                fontBook
            }
            TypefaceHelper.Type.Demi -> {
                if (fontDemi == null) {
                    fontDemi = Typeface.createFromAsset(context.assets, "FuturaPTDemi.otf")
                }
                fontDemi
            }
        }
    }
}
